catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ImportCommand.java

```diff
@@ -0,0 +1,269 @@
+
+    @Reference
+    private StorageProvider storageProvider;
+
+    @Argument(name = "Import File", description = "The file to import", index = 0, multiValued = false, required = true)
+    String importFile;
+
+    @Option(name = "--skip-signature-verification", required = false, multiValued = false, description =
+            "Exports the data but does NOT sign the resulting zip file. "
+                    + "WARNING: This file will not be able to be verified on import for integrity and authenticity.")
+    boolean unsafe = false;
+
+    @Option(name = "--force", required = false, aliases = {
+            "-f"}, multiValued = false, description = "Do not prompt")
+    boolean force = false;
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        int metacards = 0;
+        int content = 0;
+        int derivedContent = 0;
+        ZipValidator zipValidator = initZipValidator();
+        File file = initImportFile(importFile);
+        InputTransformer transformer = getServiceByFilter(InputTransformer.class,
+                String.format("(%s=%s)",
+                        "id",
+                        DEFAULT_TRANSFORMER_ID)).orElseThrow(() -> new CatalogCommandRuntimeException(
+                "Could not get " + DEFAULT_TRANSFORMER_ID + " input transformer"));
+
+        if (unsafe) {
+            if (!force) {
+                console.println("This will import data with no check to see if data is modified/corrupt. Do you wish to continue?");
+                String input = getUserInputModifiable().toString();
+                if (!input.matches("^[yY][eE]?[sS]?$")) {
+                    console.println("ABORTED IMPORT.");
+                    return null;
+                }
+            }
+            SecurityLogger.audit("Skipping validation check of imported data. There are no "
+                    + "guarantees of integrity or authenticity of the imported data."
+                    + "File being imported: {}", importFile);
+        } else {
+            if (!zipValidator.validateZipFile(importFile)) {
+                throw new CatalogCommandRuntimeException("Signature on zip file is not valid");
+            }
+        }
+        SecurityLogger.audit("Called catalog:import command on the file: {}", importFile);
+        console.println("Importing file");
+        Instant start = Instant.now();
+        try (InputStream fis = new FileInputStream(file);
+                ZipInputStream zipInputStream = new ZipInputStream(fis)) {
+            ZipEntry entry = zipInputStream.getNextEntry();
+
+            while (entry != null) {
+                String filename = entry.getName();
+
+                if (filename.startsWith("META-INF")) {
+                    entry = zipInputStream.getNextEntry();
+                    continue;
+                }
+
+                String[] pathParts = filename.split("\\" + File.separator);
+                String id = pathParts[ID];
```

Comment:
I am not a fan of asserts, but since there isn't a way to guarantee they are importing a valid file i will add it.